### DISCUSSION TITLE
how to debug (latency) zigbee problems?

### DISCUSSION THREAD

**Discussion by bartplessers** [Koenkk/zigbee2mqtt#8985] [General]
# My Problem
Last week, I noticed that my lights were reacting very slow on detected motion. All sensors and lights are connected to Z2M and automatization is done with homeassistant.
First, I thought that it was a problem with hass, but than I noticed that the zigbee messages from the sensor came with sometimes more than 10 seconds delay on my mqtt-broker. Once the message was received, the automation ran and lights turned on as expected. So it was not a problem of hass, but probably a problem with Z2M.

So I tried to pinpoint the problem and focused on the sensor: disconnected from zigbee network, reconnected again. Rebooted, restarted zigbee2mqtt, etc... but problem persisted. Even worse: I noticed the same problem with some of my other sensors....

Than I found out: when a motion sensor was triggered, and if I manually triggered another sensor (in my case I clicked on my ikea remote), than the mqtt message of the occupancy sensor appeared immediately after touching my switch... 
It seems that the mqtt-message of the occupancy sensor was "stuck somewhere on my zigbee network" and "released when something else happened". 
 
My biggest frustration now: I intended to investigate this further, but updated my RPI today. Problem seems to be disappeared, but I'm sure the updates had nothing to do with hass or Z2M. So I can reproduce the problem right now :-(

# My Question(s)
How can I debug such problems?
The LQI of my sensor was OK, it's the only thing I can see right now.
My Z2M had  log level "info", so I only saw the messages coming in 10 seconds after occupancy occurred.
Can I do a traceroute to see the way the information is delivered to my broker?
How is the mesh build? The sensor was only 1.5m away from my light, but seems to be connected with the Zigbee stick directly. Same for other sensors: sometimes I see a low LQI with a sensor connected to my coordinator, while it's only 2m away from a router that has a LQI. Why does this sensor stick with the coordinator and not with the closer light? 

Right now, the Z2M is a black box for my and I had some problems with latency in the past (i.e. #7163), so it's really hard to build my whole lightning network on this technology if I can not resolve such problems. Other devices (sonoff, tasmota, ...) work on "normal" network technology which is much more transparent for me. 

# hardware
Zigbee2MQTT version: 1.21.1-1
Adapter hardware: zzh (CC2652R Stick) - Electrolama
Adapter firmware version: {'meta': {'maintrel': 1, 'majorrel': 2, 'minorrel': 7, 'product': 1, 'revision': 20210430, 'transportrev': 2}, 'type': 'zStack3x0'}
hardware: RPI4
hass, Zigbee2MQTT + MQTT broker running in docker

Sensors that gave problems
Xiaomi RTCGQ01LM
Xiaomi MCCGQ11LM

**Comment by matejdro** [comment]
Have you gotten anywhere with that? I think I'm experiencing similar issues.

**Reply by matejdro** [reply]
I think I found the cause: https://github.com/Koenkk/zigbee2mqtt/issues/8663

Upgrading the kernel solved the issue.
